### ISSUE TITLE
How to get predictions without background

### ISSUE THREAD

**Issue by IamSVP94** [open-mmlab/mmsegmentation#3411]
I need to do image segmentation **WITHOUT** background. I have a masks png images like this:
![image](https://github.com/open-mmlab/mmsegmentation/assets/58026014/9769be25-2dd5-448e-bbfc-8a39e6d3229c)
My image unique values: `[0 - background, 1 - smoke_cat_1, 2 - smoke_cat_2]`
```
@DATASETS.register_module()
class SmokeCustomDataset(BaseSegDataset):
    """
    Smoke Custom dataset.
    """
    METAINFO = dict(
        classes=('smoke_cat_1', 'smoke_cat_2'),
        palette=[[0, 0, 255], [0, 255, 0]],
    )

    def __init__(self,
                 img_suffix='.jpg',
                 seg_map_suffix='_mask.png',
                 reduce_zero_label=True,  # remove background
                 **kwargs) -> None:
        super(SmokeCustomDataset, self).__init__(
            img_suffix=img_suffix,
            seg_map_suffix=seg_map_suffix,
            reduce_zero_label=reduce_zero_label,
            **kwargs
        )
```
After that I try to train NN with the next custom config:
```
# ----------    COMMON SETTINGS     ----------
dataset_type = 'SmokeCustomDataset'
data_root = 'data/SMOKECUSTOM_1D/'  # dataset dir
num_classes = 2

max_epochs = 100
crop_size = (512, 512)

train_batch_size = 8
learning_rate = 0.003
reduce_zero_label = True
work_dir = './work_dirs/smokecustom_fpn_r50'
num_workers = 15
log_interval = 200
by_epoch = True
# ==========    /COMMON SETTINGS    ==========

_base_ = [
    # ----------    MODEL SETTINGS      ----------
    # path to model config
    'mmdeploy/mmsegmentation/configs/_base_/models/fpn_r50.py',
    # ==========    /MODEL SETTINGS     ==========
]

# ----------    RUNTIME SETTINGS    ----------
default_scope = 'mmseg'
env_cfg = dict(
    cudnn_benchmark=True,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'),
)
vis_backends = [
    dict(type='LocalVisBackend'),
    dict(type='TensorboardVisBackend'),  # added Tensorboard logging
]
visualizer = dict(type='SegLocalVisualizer', vis_backends=vis_backends, name='visualizer')
log_processor = dict(by_epoch=False)
log_level = 'INFO'
load_from = None
resume = False

tta_model = dict(type='SegTTAModel')
# ==========    /RUNTIME SETTINGS   ==========

# ----------    SCHEDULER SETTINGS  ----------
# optimizer
optimizer = dict(type='AdamW', lr=learning_rate, weight_decay=0.0005)
optim_wrapper = dict(type='OptimWrapper', optimizer=optimizer, clip_grad=None)
# learning policy
param_scheduler = [
    # dict(type='LinearLR', by_epoch=False, start_factor=0.1, begin=0, end=2),  # custom
    dict(
        type='PolyLR',
        eta_min=1e-4,
        power=0.9,
        begin=0,  # previous end + 1
        end=max_epochs - 1,
        by_epoch=by_epoch)
]
train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=max_epochs)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', log_metric_by_epoch=by_epoch, interval=log_interval),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', by_epoch=by_epoch, interval=log_interval, save_best='mIoU'),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    visualization=dict(type='SegVisualizationHook', draw=True, interval=log_interval))
# ==========    /SCHEDULER SETTINGS ==========

# ----------    DATASET SETTINGS    ----------
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=reduce_zero_label),
    dict(
        type='RandomResize',
        scale=(2048, 512),
        ratio_range=(0.5, 2.0),
        keep_ratio=True),
    dict(type='RandomCrop', crop_size=crop_size, cat_max_ratio=0.75),
    dict(type='RandomFlip', prob=0.5, direction='horizontal'),
    dict(type='PhotoMetricDistortion'),
    dict(type='PackSegInputs')
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='Resize', scale=(2048, 512), keep_ratio=True),
    # add loading annotation after ``Resize`` because ground truth
    # does not need to do resize data transform
    dict(type='LoadAnnotations', reduce_zero_label=reduce_zero_label),
    dict(type='PackSegInputs')
]
img_ratios = [0.5, 0.75, 1.0, 1.25, 1.5, 1.75]
tta_pipeline = [
    dict(type='LoadImageFromFile', backend_args=None),
    dict(
        type='TestTimeAug',
        transforms=[
            [
                dict(type='Resize', scale_factor=r, keep_ratio=True)
                for r in img_ratios
            ],
            [
                dict(type='RandomFlip', prob=0., direction='horizontal'),
                dict(type='RandomFlip', prob=1., direction='horizontal')
            ], [dict(type='LoadAnnotations')], [dict(type='PackSegInputs')]
        ])
]
train_dataloader = dict(
    batch_size=train_batch_size,
    num_workers=num_workers,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        reduce_zero_label=reduce_zero_label,
        data_prefix=dict(img_path='images/train', seg_map_path='annotations/train'),
        pipeline=train_pipeline))
val_dataloader = dict(
    batch_size=1,
    num_workers=num_workers,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        reduce_zero_label=reduce_zero_label,
        data_prefix=dict(img_path='images/val', seg_map_path='annotations/val'),
        pipeline=test_pipeline))
test_dataloader = dict(
    batch_size=1,
    num_workers=num_workers,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        data_root=data_root,
        reduce_zero_label=reduce_zero_label,
        data_prefix=dict(img_path='images/test', seg_map_path='annotations/test'),
        pipeline=test_pipeline))

val_evaluator = dict(type='IoUMetric', iou_metrics=['mIoU', 'mFscore'])
test_evaluator = val_evaluator.copy()

# changing default model settings
data_preprocessor = dict(size=crop_size)
norm_cfg = dict(type='SyncBN', requires_grad=True)
model = dict(
    data_preprocessor=data_preprocessor,
    decode_head=dict(num_classes=num_classes),
    # auxiliary_head=dict(num_classes=num_classes),  # deeplabv3_r50-d8
    auxiliary_head=dict(  # custom for fpn_r50
        type='FCNHead',
        in_channels=256,
        in_index=2,
        channels=8,
        num_convs=1,
        concat_input=False,
        dropout_ratio=0.1,
        num_classes=num_classes,
        norm_cfg=norm_cfg,
        align_corners=False,
        loss_decode=dict(type='CrossEntropyLoss', use_sigmoid=False, loss_weight=0.4),
    ),
)
# ==========    /DATASET SETTINGS   ==========
```
In a during of train I have not very bad metrics

![image](https://github.com/open-mmlab/mmsegmentation/assets/58026014/14d4b4a0-46f5-4e3a-a7e7-077f91ff5238)

But at the same time I have a bad visual predictions:

![image](https://github.com/open-mmlab/mmsegmentation/assets/58026014/ca256a82-7dc3-42c3-a8df-ebd5c31b0571)

What I need to do for good predictions?!

**Comment by Saillxl** [comment]
How did you solve it, thank you
